src/pages/Feed.jsx

```diff
@@ -4,40 +4,63 @@ import Error from "./Error";
 import { Loader } from "lucide-react";
 import InfiniteScroll from "react-infinite-scroll-component";
 import { getArticles, fetchArticles } from "../lib/api";
-import { useParams } from "react-router-dom";
+import { Link, useParams, useNavigate, Navigate } from "react-router-dom";
+import Dropdown from "../components/Dropdown";
+
 const Feed = () => {
+  const [sortBy, setSortBy] = useState("created_at");
+  const [orderBy, setOrderBy] = useState("desc");
   const [page, setPage] = useState(2);
   const [articles, setArticles] = useState([]);
   const [error, setError] = useState(null);
   const [hasMore, setHasMore] = useState(true);
   const params = useParams();
-  const topicTitle = params.topic ? params.topic : "all";
+  const navigate = useNavigate();
+  const pageTitle = params.topic ? params.topic : "all";
+  const topic = params.topic ? params.topic : "";
   useEffect(() => {
-    getArticles(params.topic)
+    getArticles(topic, sortBy, orderBy)
       .then((data) => setArticles(data))
       .catch((err) => setError(err));
-  }, []);
+  }, [sortBy, orderBy]);
 
   const fetchData = async () => {
-    const articlesFromServer = await fetchArticles(page,params.topic);
+    const articlesFromServer = await fetchArticles(
+      page,
+      topic,
+      sortBy,
+      orderBy,
+    );
 
     setArticles([...articles, ...articlesFromServer]);
     if (articlesFromServer.length === 0 || articlesFromServer.length < 5) {
       setHasMore(false);
     }
     setPage(page + 1);
   };
+  const handleSort = (e) => {
+    setSortBy(e.target.value);
+    params.topic
+      ? navigate(`/t/${topic}?sort=${e.target.value}`)
+      : navigate(`?sort=${e.target.value}`);
+  };
+
+  const handleOrder = (e) => {
+    setOrderBy(e.target.value);
+    params.topic
+      ? navigate(`/t/${topic}?sort=${sortBy}&order_by=${e.target.value}`)
```

Comment:
You could handle this without navigating, just by setting the search params and using a useEffect to refetch data given the search terms